test/e2e/specs/editor/various/shortcut-focus-toolbar.spec.js

```diff
@@ -0,0 +1,155 @@
+/**
+ * WordPress dependencies
+ */
+const { test, expect } = require( '@wordpress/e2e-test-utils-playwright' );
+
+const documentToolbarButton = ( page ) => {
+	return page.getByRole( 'button', {
+		name: 'Toggle block inserter',
+		exact: true,
+	} );
+};
+
+const documentToolbarTooltip = ( page ) => {
+	return page.locator( 'text=Toggle block inserter' );
+};
+
+const blockToolbarButton = ( page ) => {
+	return page.getByRole( 'button', { name: 'Paragraph', exact: true } );
+};
+
+const useSelectMode = async ( page ) => {
+	await page.keyboard.press( 'Escape' );
+};
+
+const moveToToolbarShortcut = async ( pageUtils ) => {
+	await pageUtils.pressKeys( 'alt+F10' );
+};
+
+test.describe( 'Focus toolbar shortcut (alt + F10)', () => {
+	test.beforeEach( async ( { admin } ) => {
+		await admin.createNewPost();
+	} );
+
+	test( 'Focuses correct toolbar in default view options in edit mode', async ( {
+		editor,
+		page,
+		pageUtils,
+	} ) => {
+		// Test: Focus the top level toolbar from title
+		await moveToToolbarShortcut( pageUtils );
+		await expect( documentToolbarButton( page ) ).toBeFocused();
+
+		// Test: Focus document toolbar from empty block
+		await editor.insertBlock( { name: 'core/paragraph' } );
+		await moveToToolbarShortcut( pageUtils );
+		await expect( documentToolbarButton( page ) ).toBeFocused();
+
+		// Test: Focus block toolbar from block content when block toolbar isn't visible
+		await editor.insertBlock( { name: 'core/paragraph' } );
+		await page.keyboard.type(
+			"Focus to block toolbar when block toolbar isn't visible"
+		);
+		await moveToToolbarShortcut( pageUtils );
+		await expect( blockToolbarButton( page ) ).toBeFocused();
+		await expect( documentToolbarTooltip( page ) ).not.toBeVisible();
```

Comment:
Initially these were separate tests. I combined the tests into groups based into view and mode instead. This cuts the test execution time down by about half, (18s for running all the tests vs 9s for the current setup). This is recommended by Playwright Best Practices of "write fewer tests but longer tests": https://playwright.dev/docs/best-practices#write-fewer-tests-but-longer-tests. I think the time saved per commit is worth having more assertions per test.